src/pages/createOrganization/createOrganization.js

```diff
@@ -0,0 +1,233 @@
+import React, { Component } from "react";
+import { Container, Row, Col, Form, Button } from "react-bootstrap";
+import { connect } from "react-redux";
+import { createOrg } from "../../reducers/orgSlice";
+
+class CreateOrganization extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "",
+      nameError: "",
+      shortDescription: "",
+      shortDescriptionError: "",
+      email: "",
+      emailError: "",
+      website: "",
+      websiteError: "",
+      isFormInvalid: true,
+      formSubmissionError: "",
+    };
+  }
+
+  onFieldChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value }, () => {
+      this.checkFieldValidation(name);
+    });
+  };
+
+  checkFieldValidation = (field) => {
+    switch (field) {
+      case "name": {
+        if (this.state.name.length < 3) {
+          this.setState({
+            nameError: "Name must be atleast 3 characters long",
+          });
+        } else
+          this.setState({
+            nameError: null,
+          });
+        break;
+      }
+      case "email": {
+        if (
+          this.state.email.length < 255 &&
+          this.state.email.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i)
+        ) {
+          this.setState({
+            emailError: null,
+          });
+        } else {
+          this.setState({
+            emailError: "Please enter a valid email address",
+          });
+        }
+        break;
+      }
+      case "website": {
+        if (
+          //regex to be added
+          this.state.website.match()
+        ) {
+          this.setState({
+            websiteError: null,
+          });
+        } else {
+          this.setState({
+            websiteError: "Please enter a valid website URL",
+          });
+        }
+        break;
+      }
+      case "shortDescription": {
+        if (this.state.shortDescription.length < 5) {
+          this.setState({
+            shortDescriptionError: "Short Description must be atleast 5 characters long",
+          });
+        } else
+          this.setState({
+            shortDescriptionError: null,
+          });
+        break;
+      }
+    }
+    return;
+  };
+
+  onFormSubmit = (e) => {
+    e.preventDefault();
+    this.props.createOrg(
+      this.state.name,
+      this.state.shortDescription,
+      this.state.email,
+      this.state.website,
+    );
+  };
+
+  static getDerivedStateFromProps(props, state) {
```

Comment:
I think this can also be done in compoenntDidUpdate as it contains both prevState and prevProps, so you can compare this.props with prevProps or state to get the form error.

Reply:
My bad! Missed this while refactoring (i wasn't aware earlier that componentDidUpdate reacts to prop changes as well)